e3db/sodium_crypto.py

```diff
@@ -108,12 +108,13 @@ def encrypt_file(self, plaintext_filename, key):
         edk = self.encrypt_secret(key, dk, edkN)
         edk = edk[len(edkN):]
         encrypted_length = 0
-        header = "{0}.{1}.{2}.".format(FILE_VERSION, BaseCrypto.base64encode(edk), BaseCrypto.base64encode(edkN))
+        header = "{0}.{1}.{2}.".format(FILE_VERSION, BaseCrypto.base64encode(edk).decode("utf-8"), BaseCrypto.base64encode(edkN).decode("utf-8"))
         # last field is encrypted data
 
-        encrypted_file_handle.write(header)
-        m.update(header)
-        encrypted_length += len(header)
+        header_bytes = BaseCrypto.to_bytes(header)
+        m.update(header_bytes)
```

Comment:
Why are `m.update` and the header `.write` switch here?

Reply:
This is a remnant of trying to wrap my head around something. There is no difference though, at that point those two lines are unrelated.

Reply:
It doesn't matter if you update the MD5 checksum state then write the state to file, or if you write the state to file and then update the MD5 checksum state. This seems fine to me.